File: java/src/dev/csn/Mor.java
package dev.csn;

import java.util.HashMap;

public class Mor {
    public HashMap<Integer, ListItem> ids;
    Mor(Aor src) {
        ids = new HashMap<Integer, ListItem>();

        for (var item : src.items) {
            ids.put(item.itemId, item);
        }
    }

    public boolean selected(int id) {
        return ids.get(id).selected;
    }

}

File: kotlin/src/main/kotlin/Mor.kt
class Mor(aor: Aor) {
    val ids: HashMap<Int, ListItem>

    init {
        ids = HashMap(aor.size)
        for(item in aor.items) {
            ids[item.id] = item
        }
    }

    fun selected(id: Int): Boolean {
        return ids[id]?.selected ?: false
    }
}